design/security-model.md

```diff
@@ -0,0 +1,360 @@
+
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: rook-system
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: rook-system
+  namespace: rook-system
+```
+
+The `rook-system` service account is responsible for launching all pods, services, daemonsets, etc. for Rook and should have enough privilege to do and nothing more. I've not audited all the RBAC rules but a good tool to do is [here](https://github.com/liggitt/audit2rbac). For example:
+
+```
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: rook-system
+rules:
+- apiGroups: [""]
+  resources: ["pods", "services", "configmaps"]
+  verbs: [ "get", "list", "watch", "patch", "create", "update, "delete" ]
+- apiGroups: ["extensions"]
+  resources: ["deployments", "daemonsets", "replicasets"]
+  verbs: [ "get", "list", "watch", "patch", "create", "update, "delete" ]
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: [ "get", "list", "watch", "patch", "create", "update, "delete" ]
+- apiGroups: ["rook.io"]
+  resources: ["*"]
+  verbs: [ "*" ]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: rook-system
+  namespace: rook-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: rook-system
+subjects:
+- kind: ServiceAccount
+  name: rook-system
+  namespace: rook-system
+```
+
+Notably absent here are privileges to set other RBAC rules read cluster-wide secrets, etc. Because the admin created the `rook-system` namespace and service account they are free to set policies on them using PSP or namespace quotas. Similar to the ones defined [here](https://github.com/rook/rook/blob/master/Documentation/kubernetes.md#rbac-for-podsecuritypolicies)).
+
+Also note that while we use a `ClusterRole` for rook-system we only use a `RoleBinding` to grant it access to the `rook-system` namespace. It does not have cluster-wide privileges.
+
+When creating a Rook cluster the cluster admin will continue to define the namespace and cluster CRD as follows:
+
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: mycluster
+---
+apiVersion: rook.io/v1alpha1
+kind: Cluster
+metadata:
+  name: myrookcluster
+  namespace: mycluster
+	...
+```
+
+In addition we will require that the cluster-admin define a service account and role binding as follows:
+
+```
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: rook-cluster
+  namespace: mycluster
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: rook-ceph-osd
+  namespace: mycluster
+rules:
+- apiGroups: [""]
+  resources: ["configmaps"]
+  verbs: [ "get", "list", "watch", "create", "update", "delete" ]
```

Comment:
Does the cluster-admin need to define the Role? Are there any changes to that role the cluster-admin could possibly want to make w/out knowledge about rook internals?